internal/resources/kubeconfig.go

```diff
@@ -219,6 +220,16 @@ func (r *KubeconfigResource) mutate(ctx context.Context, tenantControlPlane *kam
 				return kcErr
 			}
 
+			// Post-process the kubeconfig to use the public API server address for
+			// controller manager and scheduler components instead of localhost/IP address
+			if strings.Contains(r.KubeConfigFileName, "controller-manager") || strings.Contains(r.KubeConfigFileName, "scheduler") {
+				kubeconfig, kcErr = r.replaceServerURLWithPublicAddress(kubeconfig, tenantControlPlane)
+				if kcErr != nil {
+					logger.Error(kcErr, "cannot replace server URL in kubeconfig")
+					return kcErr
+				}
+			}
```

Comment:
Unless I'm missing something, this is not required: Kamaji leverages the concept of using loopback to communicate with the API Server; otherwise, we would end up with extra network hops with potential egress traffic.